### ISSUE TITLE
Please review licensing issues with embedded rhino

### ISSUE THREAD

**Issue by blickly** [google/closure-compiler#273]
This issue was imported from Closure Compiler's previous home at http://closure-compiler.googlecode.com

The original discussion is archived at:
http://blickly.github.io/closure-compiler-issues/#935

**Comment by aharui** [comment]
The README still says that Rhino is under Netscape Public License.  Is this still true?  Can you upgrade to a more recent Rhino which is MPL 2.0?  I believe there are Apache License compatibility issues with NPL.

**Comment by concavelenz** [comment]
Unfortunately not.  What we have left of rhino is our significantly modified version of the internal AST: two files effectively Node.java and Token.java (and their test files).  The rest is code we added (the type system) and have ownership of.  I'm not really sure what we need to do to get rid of the license dependency.

**Comment by aharui** [comment]
I am not a lawyer, but my understanding is that every line of code is under some license or another.  If there are lines of code in those files from the initial commit from Rhino, then the Netscape license dependency still stands.  I believe you could "start over" by grabbing versions of those files after they were re-licensed to MPL and re-add your changes to those files.

**Comment by aharui** [comment]
I examined source code history for Node.java and Token.java in the Rhino repo at GitHub.  The changes in those files between the version your source code history says was the starting point and when Rhino switched to MPL are minimal, and only consist of new code.  I have been told that you could simply declare your Rhino dependencies to now be MPL, since Mozilla converted all of the lines of code you took to MPL, which would make your compiler usable as a dependency by projects at the Apache Software Foundation (which is why I'm bothering you on this topic).  Please change your README.md so we can use Closure Compiler in our project.

**Comment by concavelenz** [comment]
@MatrixFrog was this resolved?

**Comment by MatrixFrog** [comment]
No, I'm working on it. It will be easier to do after some internal infrastructure changes land, so I'm kind of stalling, at the moment, unfortunately.
